public/less/d2h.less

```diff
@@ -1,218 +1,64 @@
-.d2h-code-side-line ins {
-  display: inline-block;
-  margin-top: -1px;
-  text-decoration: none;
-  background-color: rgba(255, 255, 255, 0.6);
-}
-
-.d2h-code-line-prefix {
-  display: inline-flex;
-  margin-right: 10px;
-  -moz-user-select: none;
-  -ms-user-select: none;
-  -webkit-user-select: none;
-  user-select: none;
+.d2h-cntx {
+  color: rgba(255, 255, 255, 0.3);
 }
 
-.line-num1 {
-  display: inline-block;
-  float: left;
-  width: 30px;
-  overflow: hidden;
-  text-overflow: ellipsis;
+.d2h-info {
+  background-color: transparent;
+  color: rgba(255, 255, 255, 0.3);
 }
 
-.line-num2 {
-  display: inline-block;
-  float: right;
-  width: 30px;
-  overflow: hidden;
-  text-overflow: ellipsis;
+.d2h-code-side-emptyplaceholder,
+.d2h-emptyplaceholder {
+  background-color: transparent;
 }
 
-.d2h-code-linenumber {
-  position: static;
-  width: 52px;
-  min-width: 65px;
-  text-align: center;
+.d2h-code-linenumber, .d2h-code-side-linenumber {
   background-color: #727a83;
-  color: rgba(0, 0, 0, 0.5);
-  border-width: 0 1px 0 1px;
-  cursor: pointer;
-  overflow: hidden;
-  text-overflow: ellipsis;
   border-left: 1px solid rgba(0, 0, 0, 0.34);
   border-right: 1px solid rgba(0, 0, 0, 0.34);
-  -moz-user-select: none;
-  -ms-user-select: none;
-  -webkit-user-select: none;
-  user-select: none;
-}
-
-.d2h-cntx {
-  color: rgba(255, 255, 255, 0.3);
 }
 
 .d2h-code-side-linenumber {
-  position: static;
-  display: table-cell;
-  width: 52px;
-  min-width: 65px;
-  text-align: center;
-  height: 100%;
-  background-color: rgba(114, 122, 131, 1);
-  color: rgba(0, 0, 0, 0.5);
-  border-width: 0 1px 0 1px;
-  cursor: pointer;
-  overflow: hidden;
-  text-overflow: ellipsis;
-  border-left: 1px solid rgba(0, 0, 0, 0.34);
-  border-right: 1px solid rgba(0, 0, 0, 0.34);
-  -moz-user-select: none;
-  -ms-user-select: none;
-  -webkit-user-select: none;
-  user-select: none;
+  padding-right: 0.5em;
 }
 
-.d2h-del {
+.d2h-file-diff .d2h-del.d2h-change, .d2h-del {
   background-color: #E86756;
   color: #FFFFFF;
-
-  &.d2h-code-side-linenumber {
-    color: rgba(0, 0, 0, 0.5);
-  }
 }
 
-.d2h-ins {
+.d2h-file-diff .d2h-ins.d2h-change, .d2h-ins {
   background-color: #66F27B;
   color: #000000;
+}
 
-  &.d2h-code-side-linenumber {
-    color: rgba(0, 0, 0, 0.5);
-  }
+.d2h-code-line del, .d2h-code-side-line del {
+  background-color: rgba(255, 255, 255, 0.2);
 }
 
-.d2h-info {
-  color: rgba(255, 255, 255, 0.3);
+.d2h-code-line ins, .d2h-code-side-line ins {
+  background-color: rgba(255, 255, 255, 0.6);
 }
 
-.d2h-code-line-ctn {
-  .word-wrap & {
-    white-space: -moz-pre-wrap;  /* Mozilla, since 1999 */
-    white-space: -webkit-pre-wrap; /*Chrome & Safari */
-    white-space: -pre-wrap;      /* Opera 4-6 */
-    white-space: -o-pre-wrap;    /* Opera 7 */
-    white-space: pre-wrap;       /* css-3 */
-  }
+.d2h-code-linenumber, .d2h-code-side-linenumber, .d2h-code-line-prefix {
+  -moz-user-select: none;
+  -ms-user-select: none;
```

Comment:
These don't seem to be necessary any more:
https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#Browser_compatibility

Reply:
Only in the [diff2html UI](https://github.com/rtfpessoa/diff2html#diff2htmlui-usage) which we are not using. Also it seems a bit buggy. I have opened https://github.com/rtfpessoa/diff2html/issues/303 for that.